src/docker.ts

```diff
@@ -37,24 +40,53 @@ const saveDockerImages = async (): Promise<void> => {
         "read-only option was selected."
     );
   } else {
-    const preexistingImages = getState(DOCKER_IMAGES_LIST).split("\n");
-    info("Listing Docker images.");
-    const images = await execBashCommand(LIST_COMMAND);
-    const imagesList = images.split("\n");
-    const newImages = imagesList.filter(
-      (image: string): boolean => !preexistingImages.includes(image)
-    );
-    if (newImages.length === 0) {
-      info("No Docker images to save");
+    /* Access the private method, getCacheEntry, internal to the GitHub Actions
+     * cache action.
+     */
+    interface ArtifactCacheEntry {
+      cacheKey?: string;
+      scope?: string;
+      cacheVersion?: string;
+      creationTime?: string;
+      archiveLocation?: string;
+    }
+    interface CacheWithPrivateMethod extends CacheType {
+      getCacheEntry(
+        keys: string[],
+        paths: string[]
+      ): Promise<ArtifactCacheEntry | null>;
+    }
+    const cacheWithPrivateMethod = cache as CacheWithPrivateMethod;
+    const getCacheEntry = cacheWithPrivateMethod["getCacheEntry"];
+    /* Check if a cache with our key has been saved between when we checked in
+     * loadDockerImages and now.
+     */
+    const cacheHit = Boolean(await getCacheEntry([key], [DOCKER_IMAGES_PATH]));
+    if (cacheHit) {
```

Comment:
The PR can be simplified by combining these lines into a single `else if`, thereby eliminating the indentation changes to the preexisting code.